optimum/habana/accelerate/accelerator.py

```diff
@@ -385,8 +387,9 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
                 model.forward = convert_outputs_to_fp32(new_forward)
         elif self.state.is_fp8_enabled:
             model = self.wrap_fp8(model)
-            SwitchableForwardMaker.convert(model, self.fp8_recipe_handler)
-
+            for _layer in model.base_model.model.model.layers:
+                SwitchableForwardMaker.convert(_layer, self.fp8_recipe_handler, use_activation_checkpointing=True)
+            SwitchableForwardMaker.convert(model.base_model.model.lm_head, self.fp8_recipe_handler)
```

Comment:
Shouldn't we add  lm_head (or) base_model checks as this is common code and can be used by other models.
It will be easy during upstreaming changes

Reply:
@bhargaveede when upstreaming we plan to move wrapping code to respective modeling file. Idea is that when using checkpointing every model needs to know how to wrap in fp8 also.